Sources/NIO/ChannelPipeline.swift

```diff
@@ -1555,6 +1555,20 @@ extension ChannelPipeline: CustomDebugStringConvertible {
         return desc.joined(separator: "\n")
     }
     
+    /// Returns the `ChannelHandlerContext` that belongs to a `ChannelHandler` of the given type.
+    ///
+    /// - parameters:
+    ///     - handlerType: the type of `ChannelHandler` to returns
+    public func handler<T: ChannelHandler>(type _: T.Type) -> EventLoopFuture<T> {
```

Comment:
Should we have a version of this for each of our `context` methods? I've always thought the `context` methods were pretty weird, and ideally I'd get rid of them and replace them with these versions. @normanmaurer is there a good reason not to do that?

Reply:
@Lukasa I think the `context` ones are there so you can search and later on remove without walking the pipeline. But I agree they're a bit weird, not sure if we should kill them though.